packages/react/src/Token/Token.docs.json

```diff
@@ -51,6 +51,12 @@
     {
       "name": "sx",
       "type": "SystemStyleObject"
+    },
+    {
+      "name": "removeButtonLabel",
+      "type": "string",
+      "defaultValue": "'Remove token'",
+      "description": "Label to give to the remove button for assistive technologies, such as screen readers"
```

Comment:
I think this API is becoming cumbersome because it feels like it's an after thought

We have 3 props `onRemove`, `hideRemoveButton` and `removeButtonLabel` that all affect one button and clash with each other :(

I'd love to clean this up in a backward compatible way. Do you think we could/should address that as part of this work?

Reply:
This is a good point! I had a lot more planned for this PR, but unfortunately this is so intertwined with [`TextInputWithTokens`](https://primer.style/components/text-input-with-tokens/react/alpha) it was hard to add meaningful changes without breaking something in that component 😕

I do think that we'd want to keep `removeButtonLabel` as there's no current way to change the "Remove token" `aria-label` attached on the token button now. I was planning on coming back to `Token` in https://github.com/github/primer/issues/3468, as this issue has the potential to require a lot of changes to both components.

We could remove `removeButtonLabel` and handle it later since that isn't part of any of the issues in https://github.com/github/primer/issues/3571, but an accessibility enhancement, or we could ship these changes and work towards making this component a bit cleaner and scope that into https://github.com/github/primer/issues/3468. Curious what you think though!